### ISSUE TITLE
Autocomplete for attributes/methods are not proposed with Type[] in nested objects

### ISSUE THREAD

**Issue by schperplata** [microsoft/pylance-release#714]
Issue Type: <b>Bug</b>
Feel free to rename the issue, I didn't find a better title.

VS Code does not recommend possible attributes (class static variables, `@classmethods`, `@staticmethods`) if variable is a class (not instance), type hint-ed with `Type[<class>]`, packed to some other container. Example code: 

```python
class TestClass():
    variable: str = "asd"

    @classmethod
    def cmExample(cls) -> str:
        return "example @classmethod"

    @staticmethod
    def smExample() -> str:
        return "example @staticmethod"

class Item:
    def __init__(self, model: Type[TestClass]) -> None:
        self.model = model

if __name__ == "__main__":
    item = Item(TestClass)
    # item.model.cm # <-- autocomplete does not work here...
```

While `item.mo...` is correctly recommended, its (`model`) attributes are not. For example:
![image](https://user-images.githubusercontent.com/8736111/101784251-90794100-3afb-11eb-875d-0959f8054d0d.png)
... should recommend `cmExample()`, but does not. The same goes with `variable` and `smExample()`.

Replacing `: Type[TestClass]` with `: TestClass` correctly propose mentioned attributes, but is a wrong type hint - `model` is class (reference), not instance.

Extension version: 2020.12.1
VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (6.77GB free)|
|Process Argv|W:\trunk\fitIDEA3\_fitIDEA3.code-workspace --crash-reporter-id 919e0ed9-46f3-4ad4-9e28-f35755bf8266|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.12.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020122-11-december-2020
